storage/namespace.go

```diff
@@ -130,6 +136,31 @@ func (n *dbNamespace) QueryRaw(
 	return res, nil
 }
 
+func (n *dbNamespace) QueryGrouped(
+	ctx context.Context,
+	q query.ParsedGroupedQuery,
+) (query.GroupedResults, error) {
+	retentionStartNanos := n.nowFn().Add(-n.nsOpts.Retention()).UnixNano()
+	if q.StartNanosInclusive < retentionStartNanos {
+		q.StartNanosInclusive = retentionStartNanos
+	}
+
+	res := q.NewGroupedResults()
+	shards := n.getOwnedShards()
+	for _, shard := range shards {
+		shardRes, err := shard.QueryGrouped(ctx, q)
+		if err != nil {
+			return query.GroupedResults{}, err
+		}
+		res.AddBatch(shardRes.Groups)
+		if !res.IsOrdered() && res.LimitReached() {
```

Comment:
why is the `IsOrdered` check necessary here? Isn't `LimitReached` enough signal to bail?

Reply:
The idea is that if the results are ordered, even if you have gathered enough from one shard, you still want to merge w/ the results from other shards as those may be "larger" than your current results, and as such we can't bail early for ordered results. We can however do so for unordered results.

Reply:
ahh, that makes sense. Mind leaving an inline comment? Perhaps even add a new function that encapsulates this check? since we use it in 3 other places.

Reply:
Fair enough, can add a new function `IsComplete` and clarifying comments for the function, perhaps replacing `LimitReached` too.